### DISCUSSION TITLE
Can I get this same functionality with less/cleaner code or another method?

### DISCUSSION THREAD

**Discussion by aliade-consid** [react-hook-form/react-hook-form#10827] [Q&A]
When all is checked all checkboxes should get checked and it should get unchecked if any other checkbox gets unchecked. when primary is checked all secondary chackboxes should get checked but if it gets unchecked it should not affect the secondary boxes.

[codesandbox](https://codesandbox.io/s/checkall-primary-secondary-jhfrnv)

**Comment by Moshyfawn** [comment]
What specific functionally are you referring to? What's your use-case?

**Reply by aliade-consid** [reply]
When all is checked all checkboxes should get checked and it should get unchecked if any other checkbox gets unchecked. when primary is checked all secondary chackboxes should get checked but if it gets unchecked it should not affect the secondary boxes.

**Reply by aliade-consid** [reply]
Any help is greatly appreciated!
